packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/IntentSender.java

```diff
@@ -59,6 +64,30 @@ void send(Intent intent) {
     }
   }
 
+  void showChooser(Intent intent, @NonNull List<String> ignoredPackages) {
+    Intent chooser = null;
+    intent.setPackage(null);
+    intent.setComponent(null);
+
+    // Exclude ignored packages from the displayed list.
+    if (ignoredPackages.size() > 0) {
+      ArrayList<Intent> targetIntents = calculateTargetIntents(intent, ignoredPackages);
+      if (targetIntents != null && targetIntents.size() > 0) {
+        // Create a chooser with the first element of the list,
+        // removing it from the list to avoid double displaying it in the chooser,
+        // since we add the whole list further using putExtra.
+        chooser = Intent.createChooser(targetIntents.remove(0), "");
+        chooser.putExtra(Intent.EXTRA_INITIAL_INTENTS, targetIntents.toArray(new Parcelable[] {}));
+      }
+    }
+
+    if (chooser != null) {
```

Comment:
The reason for having this method is not for some specific purpose for using it. This method appeared to simplify the `buildIntent` method by removing the corresponding `showChooser` parameter from it. This parameter appeared to save the previous method behavior. Earlier, when calling the `send` method of this class, the default chooser might appear or not, depending on whether the default application is set to resolve this intent. That is, the appearance of the chooser was not controlled in any way by the developer (unless explicitly specifying a package so as not to show the chooser).

When I implemented ignoring the specified packages, there were two options: use any available activity to resolve this intent (using the `intent.setPackage` method), if it is not in the ignore list - in this case, the chooser will never appear because the activity is already selected. Another option was to show the chooser with all available activities except ignored. Therefore, this `showChooser` method appeared, so that the developer can explicitly choose this method to call, if he prefers to show the chooser.

If the chooser is null, then no available activities were found and the same thing will happen as if we called the `send` method for this intent.

What do you think about the need for this method or its name?